js/views/modals/listingDetail/Listing.js

```diff
@@ -103,6 +121,81 @@ export default class extends BaseModal {
     }
   }
 
+  onClickGotoPhotos() {
+    this.gotoPhotos();
+  }
+
+  gotoPhotos() {
+    this.$photoSection.velocity(
+      'scroll',
+      {
+        duration: 500,
+        easing: 'easeOutSine',
+        container: this.$el,
+      });
+  }
+
+  onClickPhotoSelect(e) {
+    this.setSelectedPhoto($(e.target).data('index'));
+  }
+
+  setSelectedPhoto(photoIndex) {
+    if (is.not.number(photoIndex)) {
+      throw new Error('Please provide an index for the selected photo.');
+    }
+    if (photoIndex < 0) {
+      throw new Error('Please provide a valid index for the selected photo.');
+    }
+    const photoCol = this.model.get('listing').toJSON().item.images;
+    const photoHash = photoCol[photoIndex].original;
+    const phSrc = app.getServerUrl(`ipfs/${photoHash}`);
+
+    this.$photoSection.cropit('imageSrc', phSrc);
+    this.$photoSelected.attr('data-index', photoIndex);
+  }
+
+  setActivePhotoThumbnail(thumbIndex) {
+    if (is.not.number(thumbIndex)) {
+      throw new Error('Please provide an index for the selected photo thumbnail.');
+    }
+    if (thumbIndex < 0) {
+      throw new Error('Please provide a valid index for the selected photo thumbnail.');
+    }
+    this.$photoRadioBtns.prop('checked', false).eq(thumbIndex).prop('checked', true);
+  }
+
+  onClickPhotoPrev() {
+    let targetIndex = parseInt(this.$photoSelected.attr('data-index'), 10) - 1;
+    const imagesLength = parseInt(this.model.get('listing').toJSON().item.images.length, 10);
+
+    targetIndex = targetIndex < 0 ? imagesLength - 1 : targetIndex;
+    this.setSelectedPhoto(targetIndex);
+    this.setActivePhotoThumbnail(targetIndex);
+  }
+
+  onClickPhotoNext() {
+    let targetIndex = parseInt(this.$photoSelected.attr('data-index'), 10) + 1;
+    const imagesLength = parseInt(this.model.get('listing').toJSON().item.images.length, 10);
+
+    targetIndex = targetIndex >= imagesLength ? 0 : targetIndex;
+    this.setSelectedPhoto(targetIndex);
+    this.setActivePhotoThumbnail(targetIndex);
+  }
+
+  onClickFreeShippingLabel() {
+    this.gotoShippingOptions();
+  }
+
+  gotoShippingOptions() {
+    this.$shippingSection.velocity(
```

Comment:
Don't forget to import velocity:

```
import '../../../utils/velocity';
```

The only reason it's working is because the Store imports it which adds it to the jQuery object, but if the Store were to change and not needed, this module would break. Much safer for this module to also declare it as a dependency.